src/gui/attributetable/qgsdualview.cpp

```diff
@@ -865,11 +865,22 @@ void QgsDualView::showViewHeaderMenu( QPoint point )
   connect( setWidth, &QAction::triggered, this, &QgsDualView::resizeColumn );
   setWidth->setData( col );
   mHorizontalHeaderMenu->addAction( setWidth );
+
+  QAction *setWidthAllColumns = new QAction( tr( "&Set Width All..." ), mHorizontalHeaderMenu );
+  connect( setWidthAllColumns, &QAction::triggered, this, &QgsDualView::resizeAllColumns );
+  setWidthAllColumns->setData( col );
```

Comment:
This can be removed, it's not required for an action which applies to all columns
```suggestion
```

Reply:
Nyall, without that line, I noticed that the place holder value in the QInputDialog always contains the column width of the first column in the attribute table, rather than the width of the column right-clicked on, because without calling setData( col ) the col variable inside the slot method always returns 0. Having the clicked-on column width as the place holder, imho, is an advantage which would make it really easy to resize all columns based on an existing column width e.g. just right click on a column with the desired width, select the action and hit enter. So simply for that reason, I would be in favour of leaving that line in.
The other option would have been to hard code a value (like 100) as the default but I think I like the existing approach better.

The gif below actually shows the undesirable (for me) behaviour without that line.

![Peek 2021-01-29 20-58](https://user-images.githubusercontent.com/47767794/106376479-6b110100-63e1-11eb-8e4f-c65ca96ce382.gif)

Reply:
Argh - I missed that! Go ahead and discard the suggestion :)